React/Views/RCTNavigator.m

```diff
@@ -27,6 +27,17 @@ typedef NS_ENUM(NSUInteger, RCTNavigationLock) {
   RCTNavigationLockJavaScript
 };
 
+// By default the interactive pop gesture will be enabled when the navigation bar is displayed
+// and disabled when hidden
+// RCTPopGestureStateDefault maps to the default behavior (mentioned above)
+// RCTPopGestureStateEnabled will enable the gesture independent of nav bar visibility
+// RCTPopGestureStateDisabled will disable the gesture independent of nav bar visibility
+typedef NS_ENUM(NSUInteger, RCTPopGestureState) {
```

Comment:
If the behaviour is an enum on the native side, I feel like we should also expose an enum on the JS. Otherwise there's no way to reset a navigator to the default behaviour. (e.g. https://github.com/facebook/react-native/blob/master/Libraries/Modal/Modal.js#L43)